File: app/src/main/java/igrek/songbook/layout/settings/SliderController.java
package igrek.songbook.layout.settings;

import android.widget.SeekBar;
import android.widget.TextView;

import io.reactivex.subjects.PublishSubject;

public class SliderController {
	
	private SeekBar seekBar;
	private TextView label;
	private float min;
	private float max;
	PublishSubject<Float> valueSubject = PublishSubject.create();
	
	public SliderController(SeekBar seekBar, TextView label, float currentValue, float min, float max) {
		this.seekBar = seekBar;
		this.label = label;
		this.min = min;
		this.max = max;
		
		setValue(currentValue);
		updateLabel(currentValue);
		
		seekBar.setMax(1000);
		seekBar.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
			@Override
			public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
				float value = getValue();
				updateLabel(value);
				valueSubject.onNext(getValue());
			}
			
			@Override
			public void onStartTrackingTouch(SeekBar seekBar) {
			}
			
			@Override
			public void onStopTrackingTouch(SeekBar seekBar) {
			}
		});
	}
	
	
	public String generateLabelText(float value) {
		return "";
	}
	
	private void updateLabel(float value) {
		label.setText(generateLabelText(value));
	}
	
	public void setValue(float value) {
		float progress = (value - min) * seekBar.getMax() / (max - min);
		if (progress < 0)
			progress = 0;
		if (progress > 1000)
			progress = 1000;
		seekBar.setProgress((int) progress);
	}
	
	public float getValue() {
		int progress = seekBar.getProgress();
		float value = min + (max - min) * progress / seekBar.getMax();
		return value;
	}
	
	public PublishSubject<Float> getValueSubject() {
		return valueSubject;
	}
}

File: app/src/main/java/igrek/songbook/layout/slider/SliderController.kt
package igrek.songbook.layout.slider

import android.widget.SeekBar
import android.widget.TextView

import io.reactivex.subjects.PublishSubject

open class SliderController(private val seekBar: SeekBar, private val label: TextView, currentValue: Float, private val min: Float, private val max: Float) {
    var valueSubject = PublishSubject.create<Float>()
        internal set

    var value: Float
        get() {
            val progress = seekBar.progress
            return min + (max - min) * progress / seekBar.max
        }
        set(value) {
            var progress = (value - min) * seekBar.max / (max - min)
            if (progress < 0)
                progress = 0f
            if (progress > 100000)
                progress = 100000f
            seekBar.progress = progress.toInt()
        }

    init {

        value = currentValue
        updateLabel(currentValue)

        seekBar.max = 100000
        seekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
            override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
                val value = value
                updateLabel(value)
                valueSubject.onNext(value)
            }

            override fun onStartTrackingTouch(seekBar: SeekBar) {}

            override fun onStopTrackingTouch(seekBar: SeekBar) {}
        })
    }


    open fun generateLabelText(value: Float): String {
        return ""
    }

    private fun updateLabel(value: Float) {
        label.text = generateLabelText(value)
    }
}
